### PR TITLE
ensure program exits if any thread panics

### PR DESCRIPTION
**Pull request by davidkazlauskas**
@ljank

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -49,10 +49,12 @@ pub fn udp_server(chan: Sender<Event>, port: u16) {
...
 
     let stream = stream.for_each(|(_addr, msg)| {
-        let _ = chan.send(Event::UdpMessage(Vec::from(msg)));
+        let res = chan.send(Event::UdpMessage(Vec::from(msg)));
+        res.expect("Cannot send udp message to channel");
         future::ok(())
     });
-    let _ = core.run(stream);
+    let res = core.run(stream);
+    res.expect("Udp metrics server failed");
```
Comment (ljank) [inline src/server.rs]: who is `Udp metrics server`? :)

File: src/main.rs [hunk 2]
```diff
@@ -60,6 +59,11 @@ fn main() {
              args.flag_admin_host,
              args.flag_admin_port);
 
+    std::panic::set_hook(Box::new(|err| {
+        eprintln!("Panic error {:?}, exiting program.", err);
+        std::process::exit(77);
```
Comment (ljank) [inline src/main.rs:64]: why 77? is it some random exit code? :)

Comment (davidkazlauskas) [reply src/main.rs:64]: Yeah, I didn't think of anything better, and it has to be not zero, so, I thought this random number :/

File: src/server.rs (hunk 1 above)
Comment (davidkazlauskas) [reply src/server.rs]: Well, basically handle that has the port opened. Maybe socket would be a better name, will rename

Comment (davidkazlauskas) [reply src/server.rs]: @ljank fixed

**Pull request merged**